otter/generate/autograder.py

```diff
@@ -65,15 +66,18 @@ def main(tests_path, output_path, config, lang, requirements, overwrite_requirem
     else:
         otter_config = {}
     
-    if "course_id" in otter_config and "assignment_id" in otter_config:
+    if "token" not in otter_config and token is not None:
+        otter_config["token"] = token
+    elif "token" not in otter_config and "course_id" in otter_config and "assignment_id" in otter_config:
         client = APIClient()
-        if username is not None and password is not None:
-            client.log_in(username, password)
-            token = client.token
-        else:
-            token = client.get_token()
+        if token is None or "token" not in otter_config.keys():
```

Comment:
I think this if statement is not needed. This line is never executed unless the statement is true.